assets/javascripts/modules/modalDialog.js

```diff
@@ -0,0 +1,351 @@
+ *
+ * @summary An accessible modal-dialog component module consisting of:
+ * - outer container masking element, covering the page
+ * - inner container for content, with auto scrolling if content exceeds inner container height
+ * - content(s) with visibility toggle
+ * - Tested on: IE8+, MSEdge, Safari 9.0.3, Google Chrome 48.0.2564.1, Firefox 44.0.2,
+ * - Responsive resolutions: 1920x1080, 1280x800, 768x1024, 750x1334, 600x800, 320x568, 320x480
+ *
+ * @see @link{https://designpatterns.hackpad.com/Modal-dialog-boxes-ZJNqssq4Dll|Modal dialog boxes - designpatterns.hackpad.com}
+ * @see @link{https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_dialog_role|Using the dialog role}
+ * @see @link{https://www.nczonline.net/blog/2013/02/12/making-an-accessible-dialog-box/|Making an accessible dialog box - NCZOnline}
+ * @see @link{https://www.paciellogroup.com/blog/2012/05/html5-accessibility-chops-hidden-and-aria-hidden/|HTML5 Accessibility Chops: hidden and aria-hidden}
+ * @see @link{'Tab key does not select menus or buttons|https://support.mozilla.org/en-US/kb/Pressing%20Tab%20key%20does%20not%20select%20menus%20or%20buttons}
+ *
+ * @example:
+ * - include "modal-dialog" & "modal-dialog__container" in page or page template
+ * - include "modal-dialog__content" inner html markup in partial views
+ *
+ *  <div class="modal-dialog">
+ *    <div class="modal-dialog__inner">
+ *      <div id="content-1" class="modal-dialog__content" data-ga-open-event="category:Event:id" tabindex="1" role="dialog"
+ *          aria-labelledby="heading-id" aria-describedby="message-id" hidden>
+ *        <div id="message-id">
+ *          <h2 id="heading-id">content 1 heading</h2>
+ *          ....
+ *          <button role="button" class="button button--secondary" data-journey-click="category:Event:id"
+ *            data-modaldialog-action="close" tabindex="1">close</button>
+ *        </div>
+ *      </div>
+ *      <div id="content-2" class="modal-dialog__content" data-ga-open-event="category:Event:id" tabindex="1" role="dialog"
+ *        aria-labelledby="heading-id-2" aria-describedby="message-id-2">
+ *        <div id="message-id-2">
+ *          <h2 i fd="heading-id-2">content 2 heading</h2>
+ *          ...
+ *          <a href="#" data-modaldialog-action="close" tabindex="1" data-journey-click="category:Event:id">...</a>
+ *        </div>
+ *      </div>
+ *    </div>
+ *  </div>
+ *
+ *  // to display modal-dialog on a link's click event:
+ *  <a class="" href="#" data-modaldialog-action="id-selector-of-content" data-journey-click="category:Event:id">click to open</a>
+ *
+ */
+require('jquery');
+var GOVUK = require('stageprompt');
+
+module.exports = function() {
+  var $modalDialog = $('.modal-dialog'),
+    $container = $('.modal-dialog__inner'),
+    $contents = $('.modal-dialog__content'),
+    contentShowing = '.modal-dialog__content:not([hidden])',
+    $doc = $(document),
+    $body = $('body'),
```

Comment:
You don't need to use jQuery's selector engine to get `$body` and `$doc` here...

`$doc = document` and `$body = document.body`